dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -228,6 +228,7 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
             String fileKey = file.getType().toString() + file.getAbsolutePath();
             SourceFile existingFile = existingFileMap.get(fileKey);
             if (existingFileMap.containsKey(fileKey)) {
+                existingFile.setPath(file.getPath());
```

Comment:
If I understand correctly,  this only really changes old relative paths to absolute paths. 
Worth logging this and/or commenting that this can probably go away some day when all paths in the DB are absolute?

Reply:
> If I understand correctly, this only really changes old relative paths to absolute paths. Worth logging this and/or commenting that this can probably go away some day when all paths in the DB are absolute?

Even after we convert the webservice to reckon using normalized ("absolute") paths, it still might be useful to store the original/non-normalized path ("relative") to display in the UI or for error/diagnostic purposes.  If so, this code probably should stay, because in addition to the updating the original path when it changes from relative to absolute (or vice versa), it handles the case where a relative path [in a sourcefile] is changed but still refers to the same file (ex: `./foo.wdl` -> `foo.wdl`).

I wasn't sure this code site was the best place to memorialize all of this, so I wrote a ticket to discuss it and related issues:
https://ucsc-cgl.atlassian.net/browse/SEAB-5945

Reply:
I'm a bit wary of storing both the absolute and relative path in case they  get out of sync. 
But I think we can cross that road when it comes to it.